From 53053bd181105fcb9e96e318d4d321421cf22ad2 Mon Sep 17 00:00:00 2001
From: Anton Mikanovich <amikan@ilbers.de>
Date: Thu, 8 Jul 2021 18:22:51 +0300
Subject: [PATCH 1/7] rootfs: Use separate mounts lock file

MOUNTS_LOCKFILE value inside rootfs.bbclass is overwritten by later
inherited buildchroot.bbclass. We need to use other value name.

Signed-off-by: Anton Mikanovich <amikan@ilbers.de>
---
 meta/classes/rootfs.bbclass | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meta/classes/rootfs.bbclass b/meta/classes/rootfs.bbclass
index 50d6408..6ce2cf9 100644
--- a/meta/classes/rootfs.bbclass
+++ b/meta/classes/rootfs.bbclass
@@ -26,7 +26,7 @@ export LANG = "C"
 export LANGUAGE = "C"
 export LC_ALL = "C"
 
-MOUNT_LOCKFILE = "${ROOTFSDIR}.lock"
+IMAGE_MOUNT_LOCKFILE = "${ROOTFSDIR}.lock"
 
 rootfs_do_mounts[weight] = "3"
 rootfs_do_mounts() {
@@ -70,7 +70,7 @@ rootfs_do_mounts() {
                 mount --bind '${REPO_BASE_DIR}' '${ROOTFSDIR}/base-apt'
         fi
 
-        ) 9>'${MOUNT_LOCKFILE}'
+        ) 9>'${IMAGE_MOUNT_LOCKFILE}'
 EOSUDO
 }
 
@@ -103,7 +103,7 @@ rootfs_undo_mounts() {
             umount -R ${ROOTFSDIR}/proc
         mountpoint -q '${ROOTFSDIR}/dev' && \
             umount -R ${ROOTFSDIR}/dev
-        ) 9>'${MOUNT_LOCKFILE}'
+        ) 9>'${IMAGE_MOUNT_LOCKFILE}'
 EOSUDO
 }
 
-- 
2.20.1

